epic-sample-script.ps1

```diff
@@ -0,0 +1,294 @@
+        if (! [string]::IsNullOrWhiteSpace($responseStr)) {
+            Write-Output("Body : {0}" -f $responseStr)
+        }
+    }
+    elseif ($status_code -eq 0 -or $status_code -eq 501 -or $status_code -eq 503) {
+        throw "Service unavailable."
+    }
+    else {
+        WriteErrorMessage("Failure with StatusCode [{0}] and ReasonPhrase [{1}]" -f $result.StatusCode, $result.ReasonPhrase)
+        WriteErrorMessage("Error body : {0}" -f $result.Content.ReadAsStringAsync().Result)
+    }
+}
+
+function GetOrCreateMetadata($FileName) {
+    $fileHash = ComputeHashForInputFile($FileName)
+    $metaDataFileName = [System.IO.Path]::GetDirectoryName($FileName) + "\." + [System.IO.Path]::GetFileNameWithoutExtension($FileName) + "####mdata.txt"
+    if ([System.IO.File]::Exists($metaDataFileName)) {
+        # GET metadata from file
+        $metadata = [FileMetdata](Get-Content $metaDataFileName | Out-String | ConvertFrom-Json)
+        if ($metadata.FileHash -eq $fileHash) {
+            #return Appropriate Metadata
+            return $metadata
+        }
+    }
+    
+    $newmetadata = [FileMetdata]::new()
+    $newmetadata.FileHash = $fileHash
+    $newmetadata.NoOfRowsWritten = 0
+    $newmetadata.Service = $serviceName
+    return $newmetadata
+}
+
+function UpdateMetadata($FileName, $noOfRowsWritten) {
+    # Update metadata
+    $filemetaData = [FileMetdata]::new()
+    $filemetaData.FileHash = ComputeHashForInputFile($FileName)
+    $filemetaData.Service = $serviceName
+    $filemetaData.NoOfRowsWritten = $noOfRowsWritten
+    $metaDataFilePath = GetMetadataFilePath $FileName
+    $filemetaData | ConvertTo-Json -Depth 100 | Out-File $metaDataFilePath
+}
+
+function HandleObseleteMetadata($FileName) {
+    # Delete metadata which are over 30 days old
+    $timeLimit = (Get-Date).AddDays(-30)
+    $filePath = [System.IO.Path]::GetDirectoryName($FileName)
+    Get-ChildItem -Path $filePath -Recurse -Force | Where-Object { !$_.PSIsContainer -and $_.LastWriteTime -lt $timeLimit } | Where-Object { $_.Name -match '^.+\####mdata.txt$' } | Remove-Item -Force
+}
+
+function GetMetadataFilePath($FileName) {
+    return [System.IO.Path]::GetDirectoryName($FileName) + "\." + [System.IO.Path]::GetFileNameWithoutExtension($FileName) + "####mdata.txt"
+}
+
+function ComputeHashForInputFile($FileName) {
+    $stream = [System.IO.MemoryStream]::new()
+    $writer = [System.IO.StreamWriter]::new($stream)
+    $date = ([datetime](Get-ItemProperty -Path $FileName -Name LastWriteTime).lastwritetime).ToString("yyyy-MM-ddTHH:mm:ss")
+    $writer.write($FileName + $date)
+    $writer.Flush()
+    $stream.Position = 0
+    $filemetaData = Get-FileHash -InputStream $stream | Select-Object Hash
+    $stream.Dispose()
+    $writer.Dispose()
+    return $filemetaData
+}
+
+function ChunkAndSend($FileName, $linesperFile) {
+    $DirName = [System.IO.Path]::GetDirectoryName($FileName)
+    $TmpFileName = "\tmp"
```

Comment:
Shouldn't temp data be created under $env:TEMP. Would we have permission to create files in the path specified.